### DISCUSSION TITLE
env_stocktrading.py vs. env_stocktrading_np.py

### DISCUSSION THREAD

**Discussion by khanhphan1311** [AI4Finance-Foundation/FinRL#565] [Q&A] [answered]
What is the difference between env_stocktrading vs. env_stocktrading_np?
I could not find the meaning of env_stocktrading_np creation.
Please kindly help me with this question.
Thank you much.
![Screenshot_1](https://user-images.githubusercontent.com/58284749/162886032-7cd21dd7-0462-42fe-b4c3-4ed693e3ce1e.png)

**Comment by BruceYanghy** [comment, marked as answer]
Hey khanhphan1311, 

env_stocktrading is for tutorials

env_stocktrading_np is for production (paper trading/live trading), numpy processing is faster

Thanks.

**Reply by khanhphan1311** [reply]
Thank @BruceYanghy!
